### PR TITLE
[New Rule] Launch Agent Creation or Modification followed by Loading

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/695

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/macos/persistence_creation_change_launch_agents_file.toml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+description = """
+An adversary can establish persistence by installing a new launch agent that executes at login by using launchd or
+launchctl to load a plist into the appropriate directories.
+"""
+false_positives = ["Trusted applications persisting via LaunchAgent"]
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License"
+name = "Launch Agent Creation or Modification followed by Loading"
```
Comment (brokensound77) [inline rules/macos/persistence_creation_change_launch_agents_file.toml:17]:
```suggestion
name = "Launch Agent Creation or Modification and Immediate Loading"
```

File: rules/macos/persistence_creation_change_launch_agents_file.toml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    "https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPSystemStartup/Chapters/CreatingLaunchdJobs.html",
+]
+risk_score = 21
+rule_id = "082e3f8c-6f80-485c-91eb-5b112cb79b28"
+severity = "low"
+tags = ["Elastic", "Host", "macOS", "Threat Detection", "Persistence"]
+type = "eql"
+
+query = '''
+sequence by host.id with maxspan = 1m
```
Comment (brokensound77) [inline rules/macos/persistence_creation_change_launch_agents_file.toml:28]:
```suggestion
sequence by host.id with maxspan=1m
```

File: rules/macos/persistence_creation_change_launch_agents_file.toml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ [file where event.type != "deletion" and 
+  file.path : ("/System/Library/LaunchAgents/*", "/Library/LaunchAgents/*", "/Users/*/Library/LaunchAgents/*")
+ ]
+ [process where event.type in ("start", "process_started") and process.name : "launchctl" and process.args : "load"]
```
Comment (brokensound77) [inline rules/macos/persistence_creation_change_launch_agents_file.toml:32]: we should leverage case sensitivity where possible for Linux and Mac

```suggestion
  file.path in ("/System/Library/LaunchAgents/*", "/Library/LaunchAgents/*", "/Users/*/Library/LaunchAgents/*")
 ]
 [process where event.type in ("start", "process_started") and process.name == "launchctl" and process.args == "load"]
```

**Review by brokensound77 — APPROVED** [review]
A few comments, then LGTM

File: rules/macos/persistence_creation_change_launch_agents_file.toml (hunk 3 above)
Comment (Samirbous) [reply rules/macos/persistence_creation_change_launch_agents_file.toml:32]: don't think "in" works with wildcard as expected something in ("a*", "c*")

**Review by devonakerr — APPROVED** [review]

**Pull request merged**